lib/spack/spack/spec.py

```diff
@@ -1886,34 +2033,61 @@ def __init__(self):
             (r'\=',        lambda scanner, val: self.token(EQ,    val)),
             # This is more liberal than identifier_re (see above).
             # Checked by check_identifier() for better error messages.
+            (r'([\"\'])(?:(?=(\\?))\2.)*?\1',lambda scanner, val: self.token(QT, val)),
+#            (r'([\"\'])([^\1]+?)(\1)',lambda scanner, val: self.token(QT, val)),
             (r'\w[\w.-]*', lambda scanner, val: self.token(ID,    val)),
             (r'\s+',       lambda scanner, val: None)])
 
 
 class SpecParser(spack.parse.Parser):
     def __init__(self):
         super(SpecParser, self).__init__(SpecLexer())
-
+        self.previous = None
 
     def do_parse(self):
         specs = []
 
         try:
             while self.next:
+                # TODO: clean this parsing up a bit
```

Comment:
I'm suspicious this parsing could be simpler but if it works for now i think it's fine.  I would really like to allow unescaped quotes but for now this is ok.